kornia/losses/focal.py

```diff
@@ -207,9 +211,17 @@ def binary_focal_loss_with_logits(
     if input.size(0) != target.size(0):
         raise ValueError(f'Expected input batch_size ({input.size(0)}) to match target batch_size ({target.size(0)}).')
 
+    if pos_weight is None:
+        pos_weight = torch.ones(input.size(-1), device=input.device)
+    elif not isinstance(pos_weight, torch.Tensor):
```

Comment:
i suggest to use the new testing api here: https://github.com/kornia/kornia/blob/master/kornia/testing/__init__.py#L208-L228

and eventually update the existing checks

Reply:
I did it for checking 'is tensor', but when I try to replace condition of checking 'equals to number of classes' with 
`KORNIA_CHECK(input.size(-1)` == pos_weight.size(0), "Expected pos_weight equals number of classes.")`

it's problem for jit test:
`Expected a value of type 'Tensor (inferred)' for argument 'condition' but instead found type 'bool'. Inferred 'condition' to be of type 'Tensor' because it was not annotated with an explicit type.`

Reply:
I resolved this problem by adding typo for `condition` param:
`def KORNIA_CHECK(condition: bool, msg: Optional[str] = None):`

Reply:
@edgarriba Is this a valid solution?

Reply:
I think yes as soon as all test pass. Torchscript is highly sensible and tricky

Reply:
There was one fail in running test adjust.py. 

I changed `KORNIA_CHECK((factor >= 0).any(), f"Contrast factor must be positive. Got {factor}")` to `KORNIA_CHECK(any(factor >= 0), f"Contrast factor must be positive. Got {factor}")`. Python function any() outputs bool value, but output of .any() not declared.

Can you approve running tests?

Reply:
i checked that one from my side and seems fine :)